Automattic/woocommerce-payments
# Link back to evidence form from success notice

Fixes #521

Adds a link to the snackbar notice that appears after saving or submitting evidence for a dispute challenge.

The link returns the user to the "Challenge dispute" page.

Evidence saved flow:

| Before | After |
|-|-|
|![before-evidence-saved](https://user-images.githubusercontent.com/1527164/124406557-076e7380-dd85-11eb-99ae-8a8b234e5f2e.png)|![after-evidence-saved](https://user-images.githubusercontent.com/1527164/124406593-2240e800-dd85-11eb-9d67-235195299b91.png)|

Evidence submitted flow:

| Before | After |
|-|-|
|![before-evidence-submitted](https://user-images.githubusercontent.com/1527164/124406669-51575980-dd85-11eb-8d0a-3a4b07af4a0d.png)|![after-evidence-submitted](https://user-images.githubusercontent.com/1527164/124406614-3389f480-dd85-11eb-8b5a-227f6119ec8d.png)|

#### Testing instructions

* Place an order using a test card number that will simulate a dispute. You can use `4000000000000259`, or see https://stripe.com/docs/testing#disputes
* Navigate to `/wp-admin/admin.php?page=wc-admin&path=/payments/disputes`
* Click the newly created dispute to view the dispute details.
* Click the "Challenge dispute" button.
* Fill in some example dispute evidence.

**Save evidence flow**
* Click the "Save for later" button.
* Observe the snackbar notice in the bottom left-hand side of the screen with the "Return to evidence submission" link.
* Clicking the link will return you to the "Challenge dispute" page.

**Submit evidence flow**
* Click the "Submit evidence" button.
* Observe the snackbar notice in the bottom left-hand side of the screen with the "View submitted evidence" link.
* Clicking the link will return you to the "Challenge dispute" page.

**Note:** I *have not* added automated tests for these changes. I wasn't able to find existing tests for this component and introducing tests for this component might be tricky. This component contains a decent number of hooks (both state and effects) and seems like it'd be difficult to mock.

Let me know if tests are required and I'll see what I can do 😄.

-------------------
